doc/config-generic-update-rollback/SONiC_Generic_Config_Update_and_Rollback_Design.md

```diff
@@ -0,0 +1,867 @@
+| config checkpoint *filename* --verbose                   | Will save ConfigDB JSON config as a checkpoint with the name *checkpoint-name*. The CLI output will include additional details about each step executed as part of the operation. |
+| config checkpoint *filename* --dry-run --verbose         | Displays the generates commands going to be executed without running them. The CLI output will include additional details about each step executed as part of the operation.  |
+
+**checkpoint**
+
+*Command Format*
+
+config rollback <*checkpoint-name*> [--dry-run] [--verbose]
+
+| Command Option     | Purpose                                                                       |
+| ------------------ | ----------------------------------------------------------------------------- |
+|<*checkpoint-name*> | The name of the checkpoint where ConfigDB JSON config will saved under        |
+|dry-run             | Displays the generates commands going to be executed without running them.    |
+|verbose             | Provide additional details about each step executed as part of the operation. |
+
+*Command Usage*
+
+| Command                                        | Purpose                                                      |
+| ---------------------------------------------- | ------------------------------------------------------------ |
+| config rollback *checkpoint-name*              | Rolls back the ConfigDB JSON config to the config saved under *checkpoint-name* checkpoint. |
+| config rollback *filename* --dry-run           | Displays the generates commands going to be executed without running them. |
+| config rollback *filename* --verbose           | Rolls back the ConfigDB JSON config to the config saved under *checkpoint-name* checkpoint. The CLI output will include additional details about each step executed as part of the operation.
+| config rollback *filename* --dry-run --verbose | Displays the generates commands going to be executed without running them. The CLI output will include additional details about each step executed as part of the operation.  |
+
+#### 3.2.2.2 Show Commands
+
+**apply-patch**
+*Command Format*
+
+show apply-patch log [exec | verify | status]
+
+| Command                     | Purpose                                                      |
+| --------------------------- | ------------------------------------------------------------ |
+| show apply-patch log exec   | Displays a log of all the ConfigDB operations executed including<br/>those that failed. In case of a failed operation, it displays an<br/>error message against the failed operation. |
+| show apply-patch log verify | Displays a log all the ConfigDB operations that failed, <br>along with an error message. It does not display the <br/>operations that were successful. |
+| show apply-patch log status | Displays the status of last successful patch application<br/>operation since switch reboot. |
+
+**rollback**
```

Comment:
Overall it's a good start. Can we have a meeting about it because that will clarify more points and also we can share some knowledge because we have done similar stuff. Thx.

Reply:
Booked a meeting on 3/15